Rubberduck.Core/UI/ToDoItems/ToDoExplorerViewModel.cs

```diff
@@ -306,5 +306,35 @@ public void Dispose()
                 _state.StateChanged -= HandleStateChanged;
             }
         }
+
+        private string _toDoFilter;
+        public string ToDoFilter 
+        { 
+            get => _toDoFilter;
+            set
+            {
+                if (_toDoFilter != value)
+                {
+                    _toDoFilter = value;
+                    OnPropertyChanged();
+                    Items.Filter = FilterResults;
+                    OnPropertyChanged(nameof(Items));
+                }
+            }
+        }
+
+        private bool FilterResults(object marker)
+        {
+            var toDoItem = marker as ToDoItem;
+
+            if (string.IsNullOrEmpty(ToDoFilter))
+            {
+                return true;
+            }
+            
+            var passesMarkerFilter = toDoItem.Description.ToUpper().Contains(ToDoFilter.ToUpper());
+
+            return passesMarkerFilter;
+        }
```

Comment:
`passesMarkerFilter` evaluation can be simplified: 

`return string.IsNullOrEmpty(ToDoFilter) ||  toDoItem.Description.ToUpper().Contains(ToDoFilter.ToUpper());`

Reply:
`.ToUpper` may not play well with non-ANSI characters; I believe `.Contains` has an overload that takes a `StringComparison` enum that could make it case-insensitive, no?

Reply:
[String.Contains(String)](https://docs.microsoft.com/en-us/dotnet/api/system.string.contains?view=netframework-4.7) is the only overload. Were you thinking of [ToUpper(CultureInfo)](https://docs.microsoft.com/en-us/dotnet/api/system.string.toupper?view=netframework-4.7#System_String_ToUpper_System_Globalization_CultureInfo_) or [String.ToUpperInvariant()](https://docs.microsoft.com/en-us/dotnet/api/system.string.toupperinvariant?view=netframework-4.7)?

Reply:
Although `Contains` does not have an appropriate overload, `IndexOf` has. So a case-insensitive `Contains` can be achieved as `source.IndexOf(searchValue, StringComparison.CurrentCultureIgnoreCase) >= 0`. However, to actually adhere to the rules of the language selected in RD's language selection instead of the OS language, one would have to use `CurrentUICulture.CompareInfo.IndexOf(searchedString, searchString, CompareOptions.IgnoreCase) >=0`.